cmd/algorand-indexer/daemon.go

```diff
@@ -52,253 +59,343 @@ var (
+	cfg.flags.Uint32VarP(&cfg.maxAccountsLimit, "max-accounts-limit", "", 1000, "set the maximum allowed Limit parameter for querying accounts")
+	cfg.flags.Uint32VarP(&cfg.defaultAccountsLimit, "default-accounts-limit", "", 100, "set the default Limit parameter for querying accounts, if none is provided")
+	cfg.flags.Uint32VarP(&cfg.maxAssetsLimit, "max-assets-limit", "", 1000, "set the maximum allowed Limit parameter for querying assets")
+	cfg.flags.Uint32VarP(&cfg.defaultAssetsLimit, "default-assets-limit", "", 100, "set the default Limit parameter for querying assets, if none is provided")
+	cfg.flags.Uint32VarP(&cfg.maxBalancesLimit, "max-balances-limit", "", 10000, "set the maximum allowed Limit parameter for querying balances")
+	cfg.flags.Uint32VarP(&cfg.defaultBalancesLimit, "default-balances-limit", "", 1000, "set the default Limit parameter for querying balances, if none is provided")
+	cfg.flags.Uint32VarP(&cfg.maxApplicationsLimit, "max-applications-limit", "", 1000, "set the maximum allowed Limit parameter for querying applications")
+	cfg.flags.Uint32VarP(&cfg.defaultApplicationsLimit, "default-applications-limit", "", 100, "set the default Limit parameter for querying applications, if none is provided")
+
+	cfg.flags.StringVarP(&cfg.indexerDataDir, "data-dir", "i", "", "path to indexer data dir, or $INDEXER_DATA")
+	cfg.flags.BoolVar(&cfg.initLedger, "init-ledger", true, "initialize local ledger using sequential mode")
+	cfg.flags.StringVarP(&cfg.catchpoint, "catchpoint", "", "", "initialize local ledger using fast catchup")
+
+	cfg.flags.StringVarP(&cfg.cpuProfile, "cpuprofile", "", "", "file to record cpu profile to")
+	cfg.flags.StringVarP(&cfg.pidFilePath, "pidfile", "", "", "file to write daemon's process id to")
+	cfg.flags.StringVarP(&cfg.configFile, "configfile", "c", "", "file path to configuration file (indexer.yml)")
+	viper.RegisterAlias("algod", "algod-data-dir")
+	viper.RegisterAlias("algod-net", "algod-address")
+	viper.RegisterAlias("server", "server-address")
+	viper.RegisterAlias("token", "api-token")
+	viper.RegisterAlias("data-dir", "data")
+	return daemonCmd
+}
+
+func configureIndexerDataDir(indexerDataDir string) error {
+	var err error
+	if indexerDataDir == "" {
+		err = fmt.Errorf("indexer data directory was not provided")
+		logger.WithError(err).Errorf("indexer data directory error, %v", err)
+		return err
+	}
+	if _, err = os.Stat(indexerDataDir); os.IsNotExist(err) {
+		err = os.Mkdir(indexerDataDir, 0755)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "failed to configure logger: %v", err)
-			panic(exit{1})
+			logger.WithError(err).Errorf("indexer data directory error, %v", err)
+			return err
 		}
+	}
+	return err
+}
 
-		if indexerDataDir == "" {
-			fmt.Fprint(os.Stderr, "indexer data directory was not provided")
-			panic(exit{1})
+func loadIndexerConfig(indexerDataDir string, configFile string) error {
```

Comment:
I know this is a private function but can we add a comment saying BindFlagSet should be called after this and explaining why?

Reply:
Could you make a PR for this?